test/mpi/util/mtest_thread_abt.h

```diff
@@ -193,16 +193,27 @@ void MTest_init_thread_pkg(void)
 
     ABT_init(0, NULL);
 
+    /* Get the main pool of the primary execution stream.
+     *
+     * Some threads (e.g., asynchronous progress threads) can be associated with
+     * the current main pool of the primary execution stream, so we cannot
+     * update the scheduler of the primary execution stream. */
+    ret = ABT_xstream_self(&xstreams[0]);
+    MTEST_ABT_ERROR(ret, "ABT_xstream_self");
+    ret = ABT_xstream_get_main_pools(xstreams[0], 1, &pools[0]);
+    MTEST_ABT_ERROR(ret, "ABT_xstream_get_main_pools");
+
     /* Create pools */
-    for (i = 0; i < MTEST_NUM_XSTREAMS; i++) {
+    for (i = 1; i < MTEST_NUM_XSTREAMS; i++) {
         ret = ABT_pool_create_basic(ABT_POOL_FIFO, ABT_POOL_ACCESS_MPMC, ABT_TRUE, &pools[i]);
         MTEST_ABT_ERROR(ret, "ABT_pool_create_basic");
     }
 
     /* Create schedulers */
     ABT_pool my_pools[MTEST_NUM_XSTREAMS];
     num_xstreams = MTEST_NUM_XSTREAMS;
-    for (i = 0; i < num_xstreams; i++) {
+    scheds[0] = ABT_SCHED_NULL;
```

Comment:
To understand, it seems the only effective change is to avoid setting sched for the main execution stream, which I believe uses a default scheduler on `pools[0]`?

Doesn't this code runs *before* `MPI_Init_thread`?

Reply:
> To understand, it seems the only effective change is to avoid setting sched for the main execution stream, which I believe uses a default scheduler on `pools[0]`?

Previously, `scheds[0]` and `pools[0]` are created for the primary execution stream. Now the primary execution stream uses the default scheduler that has the default pool.

> Doesn't this code runs *before* `MPI_Init_thread`?

https://github.com/pmodels/mpich/blob/main/test/mpi/threads/spawn/th_taskmanager.c is the exception. This patch is for this (and potentially for other tests, too).

Reply:
I see. Is it better to fix that test?

Reply:
Um, it works with Pthreads without any problem. I hesitate to change it. Potentially this scheduler change in the original Argobots code is problematic. I believe this fix is simpler.